src/spikeinterface/preprocessing/decimate.py

```diff
@@ -0,0 +1,133 @@
+        if not isinstance(decimation_frame_start, int) or decimation_frame_step < 0:
+            raise ValueError(f"Expecting positive integer for `decimation_frame_step` arg")
+        self._decimation_frame_start = decimation_frame_start
+        resample_rate = self._orig_samp_freq / self._decimation_frame_step
+        self._sampling_frequency = resample_rate
+        # fix_dtype not always returns the str, make sure it does
+        dtype = fix_dtype(recording, None).str
+
+        BasePreprocessor.__init__(self, recording, sampling_frequency=resample_rate, dtype=dtype)
+
+        # in case there was a time_vector, it will be dropped for sanity.
+        # This is not necessary but consistent with ResampleRecording
+        for parent_segment in recording._recording_segments:
+            parent_segment.time_vector = None
+            self.add_recording_segment(
+                DecimateRecordingSegment(
+                    parent_segment,
+                    resample_rate,
+                    self._orig_samp_freq,
+                    decimation_frame_step,
+                    decimation_frame_start,
+                    dtype,
+                )
+            )
+
+        self._kwargs = dict(
+            recording=recording,
+            decimation_frame_step=decimation_frame_step,
+            decimation_frame_start=decimation_frame_start,
+        )
+
+
+class DecimateRecordingSegment(BaseRecordingSegment):
+    def __init__(
+        self,
+        parent_recording_segment,
+        resample_rate,
+        parent_rate,
+        decimation_frame_step,
+        decimation_frame_start,
+        dtype,
+    ):
+        if parent_recording_segment.t_start is None:
+            new_t_start = None
+        else:
+            new_t_start = parent_recording_segment.t_start + decimation_frame_start / parent_rate
+
+        # Do not use BasePreprocessorSegment bcause we have to reset the sampling rate!
+        BaseRecordingSegment.__init__(
+            self,
+            sampling_frequency=resample_rate,
+            t_start=new_t_start,
+        )
+        self._parent_segment = parent_recording_segment
+        self._decimation_frame_step = decimation_frame_step
+        self._decimation_frame_start = decimation_frame_start
+        self._dtype = dtype
+
+    def get_num_samples(self):
+        return len(
+            range(self._decimation_frame_start, self._parent_segment.get_num_samples(), self._decimation_frame_step)
+        )
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        # Account for offset and end when querying parent traces
+        parent_start_frame = self._decimation_frame_start + start_frame * self._decimation_frame_step
+        parent_end_frame = parent_start_frame + (end_frame - start_frame) * self._decimation_frame_step
```

Comment:
How can we be sure that this is not longer than the parent size ?
this is also related to the get_nump_samples() to be sure of ceiling vs rouding.

Reply:
It can be longer than the parent size if `end_frame > self.get_num_samples()`, but that will still return the correct output I think. 

This is related to the discussion about enforcing `end_frame <= num_samples` some time ago, not sure what was decided then?

Reply:
yes, so the majority vote was to be strict about this.

I think it is ok in this case to have end_frame larger than the parent num samples (it shouldn't throuw an error), but capping at parent segment num samples won't hurt

Reply:
I clipped start_frame/end_frame to num_samp.
Current behavior does not seem to be standardized in other preprocessors so I'm not sure exactly what to enforce here... Should I check that start_frame/end_frame are positive integers?

Reply:
This is the last point (hopefully)